### ISSUE TITLE
Attempting to add a loader results in TypeError: Cannot read property 'push' of undefined

### ISSUE THREAD

**Issue by tgorgdotcom** [SharePoint/sp-dev-docs#927]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior
Adding a loader for .woff2 files to the Sharepoint Framework webpack config

#### Observed Behavior
Error occurs when running `gulp serve`:

```
 TypeError: Cannot read property 'push' of undefined
    at Object.additionalConfiguration (/Volumes/tgorg/Work/peach/ProjectPeach/gulpfile.js:8:40)
    at ConfigureWebpackTask.executeTask (/Volumes/tgorg/Work/peach/ProjectPeach/node_modules/@microsoft/sp-build-core-tasks/lib/configureWebpack/ConfigureWebpackTask.js:432:38)
    at /Volumes/tgorg/Work/peach/ProjectPeach/node_modules/@microsoft/gulp-core-build/lib/tasks/GulpTask.js:179:32
    at ConfigureWebpackTask.GulpTask.execute (/Volumes/tgorg/Work/peach/ProjectPeach/node_modules/@microsoft/gulp-core-build/lib/tasks/GulpTask.js:171:16)
    at _executeTask (/Volumes/tgorg/Work/peach/ProjectPeach/node_modules/@microsoft/gulp-core-build/lib/index.js:363:42)
    at /Volumes/tgorg/Work/peach/ProjectPeach/node_modules/@microsoft/gulp-core-build/lib/index.js:264:59
    at process._tickCallback (internal/process/next_tick.js:103:7)
```

#### Steps to Reproduce
Added the following code to gulpfile.js right before `build.initialize(gulp);`

```
build.configureWebpack.mergeConfig({
	additionalConfiguration: (generatedConfiguration) => {
		generatedConfiguration.module.loaders.push([
			{ test: /\.woff2(\?v=[0-9]\.[0-9]\.[0-9])?$/, loader: "url-loader" }
		]);
		return generatedConfiguration;
	}
});

```

I looked through the code and I think that the problem is in `node_modules/@microsoft/sp-build-core-tasks/lib/configureWebpack/ConfigureWebpackTask.js`, line 459-468:

```
ConfigureWebpackTask.prototype._appendToBaseConfig = function (config) {
        return lodash.mergeWith({}, this._baseConfig, config, function (value1, value2) {
            if (lodash.isArray(value1) && lodash.isArray(value2)) {
                return lodash.concat(value1, value2);
            }
            else {
                return undefined;
            }
        });
    };
```

`_appendToBaseConfig` is being called in line 427 to get the config object, but for some reason it's expecting an array even though, at least from what I can gather, both `_baseConfig` and the `configAddons` var are both objects.  

Hope that helps!

**Comment by iclanton** [comment]
This is actually a documentation bug around the way we describe how to extend the Webpack config - thanks for pointing it out. We noticed this a few days ago and should have updated documentation out soon.

The issue is that the documentation describes how to extend the Webpack v1 config and the config format changed between Webpack v1 and v2, but we didn't update the documentation when we released tools with Webpack v2. If you're curious, [here's a summary of the changes between the two config formats](https://webpack.js.org/guides/migrating/#module-loaders-is-now-module-rules).

For your case, this code should work:

```JavaScript
build.configureWebpack.mergeConfig({
  additionalConfiguration: (generatedConfiguration) => {
    generatedConfiguration.module.rules.push(
      {
        test: /\.woff2(\?v=[0-9]\.[0-9]\.[0-9])?$/,
        use: {
          loader: 'url-loader'
        }
      }
    );

    return generatedConfiguration;
  }
});
```
